logdevice/ops/ldquery/tables/Directories.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#include "logdevice/ops/ldquery/tables/Directories.h"
+
+#include <queue>
+#include <folly/Conv.h>
+#include <folly/json.h>
+
+#include "../Table.h"
+#include "../Utils.h"
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/ReplicationProperty.h"
+#include "logdevice/common/configuration/UpdateableConfig.h"
+#include "logdevice/common/debug.h"
+#include "logdevice/lib/ClientImpl.h"
+
+
+namespace facebook {
+  namespace logdevice {
+    namespace ldquery {
+      namespace tables {
+
+TableColumns Directories::getColumns() const {
```

Comment:
Directories share a lot of the columns with log groups because that's how attributes are inherited. Maybe both this table and the LogGroup table can inherit from a common base that provides functions to populate and fill attribute columns?

Something like:

`TableColumns LogTreeBase::getAttributeColumns() const`
and
`void LogTreeBase::populateAttributeDate(TableData&, const LogAttributes&) const`

What do you think?